app/controllers/accounts_controller.rb

```diff
@@ -0,0 +1,55 @@
+class AccountsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :authorise_admin, only: %i[index show admin_generator]
+
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(account_params)
+      flash[:info] = "Account successfully updated"
+      redirect_to edit_account_path(params[:id])
+    else
+      flash[:error] = "error"
+      render "edit"
+    end
+  end
+
+  def admin_generator
+    @user = User.find(params[:id])
+    if @user.update(user_actions: { "admin" => { "can_administer" => true } })
+      flash[:success] = "Account successfully updated"
+      redirect_to accounts_path
+    else
+      flash[:error] = "error"
```

Comment:
we should be able to do a `@user.errors.messages` to get the error message - to test this works you could get to the screen where you wnat to make someone an admin, then in the db delete that user, then click make admin and the error message should be record not found - hmm actually no it would blow up in line 29 User.find

you could add a validation rule that user_actions need to be numeric or something and then that error message should come through